### ISSUE TITLE
\n is not converted while preview in designer page

### ISSUE THREAD

**Issue by ColinTree** [mit-cml/appinventor-sources#1099]
![image](https://user-images.githubusercontent.com/22613139/34469199-66b6dd96-ef54-11e7-992b-78d43d5059ba.png)

I think if we implement this, newe users would be easier to know this.

**Comment by ewpatton** [comment]
I think the behavior needs to be keyed off of HTMLFormat. If we are in HTMLFormat mode, then I believe the `\n` does nothing (one would use `<br>` instead). But when HTMLFormat is false, then `\n` means linebreak. Someone should double check this set of assumptions on a real device before implementing this change.

**Comment by ColinTree** [comment]
Yes this may be affected by HTMLFormat, and the problem is only at the preview page on the browser design page.

**Comment by AnujSinghal12** [comment]
I've noted that this same issue is also present in other components which show text. The ones I've found to be affected by this issue are `Button`, `TextBox`, `ListPicker`, `CheckBox`. There could be more too.

**Comment by ewpatton** [comment]
Many of these have shared base classes, so the problem may not be as prevalent as it seems. For example, MockButtonBase is the superclass of many of the button-like objects (Button, ListPicker, etc.) and so making the change there will eliminate it across a range of components. I'd definitely take a look at MockButtonBase, MockToggleBase, and MockTextBoxBase for starters.
